api/base/serializers.py

```diff
@@ -13,6 +12,27 @@
 from api.base.exceptions import InvalidQueryStringError, Conflict
 
 
+def format_relationship_links(related_link, self_link, rel_meta, self_meta):
+
+    ret = {
+        'links': {
+            'related': {
+                'href': related_link,
+                'meta': rel_meta
+            },
+            'self': {
+                'href': self_link,
+                'meta': self_meta
+            }
+        }
+    }
+
+    # Not currently returning self links
+    if not ret['links']['self']['href']:
```

Comment:
Hm, I haven't looked at Sam's PR yet, but it seems that `RelationshipField` should be responsible for correctly handling relationships that don't exist as well as empty/null relationships.

I've recently run into this use case in marshmallow-jsonapi. Relevant discussion happening here: https://github.com/marshmallow-code/marshmallow-jsonapi/pull/14

I think the following behavior is correct:

- For relationships that don't exist (e.g. node without a parent), the relationships object should not be present in the response. You may be able to use `SkipField` for this.
- For null/empty relationships, the relationship should serialize to a `null` or an empty Array.

```js
// Project without a license
"relationships": {
    "license": null
}


// User profile with no schools
"relationships": {
    "schools": []
}
```